eradiate/scenes/atmosphere/_molecules.py

```diff
@@ -0,0 +1,440 @@
+"""
+Molecular atmospheres.
+"""
+
+from __future__ import annotations
+
+import pathlib
+import tempfile
+from typing import MutableMapping, Optional
+
+import attr
+import numpy as np
+import pint
+import xarray as xr
+
+from ._core import Atmosphere, atmosphere_factory, write_binary_grid3d
+from ..phase import PhaseFunction, RayleighPhaseFunction, phase_function_factory
+from ..._util import onedict_value
+from ...attrs import AUTO, documented, parse_docs
+from ...contexts import KernelDictContext, SpectralContext
+from ...kernel.transform import map_cube, map_unit_cube
+from ...radprops.rad_profile import AFGL1986RadProfile, RadProfile, US76ApproxRadProfile
+from ...thermoprops import afgl1986, us76
+from ...thermoprops.util import (
+    compute_scaling_factors,
+    interpolate,
+    rescale_concentration,
+)
+from ...units import to_quantity
+from ...units import unit_context_kernel as uck
+from ...units import unit_registry as ureg
+
+
+@atmosphere_factory.register(
+    type_id="molecular_atmosphere", dict_constructor="afgl1986"
+)
+@parse_docs
+@attr.s
+class MolecularAtmosphere(Atmosphere):
+    """
+    Molecular atmosphere.
+
+    .. admonition:: Class method constructors
+
+       .. autosummary::
+
+          afgl1986
+          ussa1976
+    """
+
+    _thermoprops: xr.Dataset = documented(
+        attr.ib(
+            factory=us76.make_profile,
+            validator=attr.validators.instance_of(xr.Dataset),
+        ),
+        doc="Thermophysical properties.",
+        type=":class:`~xarray.Dataset`",
+    )
+
+    phase: PhaseFunction = documented(
+        attr.ib(
+            factory=lambda: RayleighPhaseFunction(),
+            converter=phase_function_factory.convert,
+            validator=attr.validators.instance_of(PhaseFunction),
+        )
+    )
+
+    has_absorption: Optional[bool] = documented(
```

Comment:
[`Optional[X]` is equivalent to `Union[X, None]`](https://docs.python.org/3/library/typing.html#typing.Optional); you don't implement this behaviour here, so your annotation should simply be:
```suggestion
    has_absorption: bool = documented(
```